packages/rs-drive-abci/src/execution/platform_events/state_transition_processing/process_raw_state_transitions/v0/mod.rs

```diff
@@ -68,197 +74,239 @@ where
         let state_transition_container =
             self.decode_raw_state_transitions(raw_state_transitions, platform_version)?;
 
-        let (
-            valid_state_transitions,
-            invalid_state_transitions,
-            invalid_execution_abci_state_transitions,
-        ) = state_transition_container.destructure();
-
         let mut processing_result = StateTransitionsProcessingResult::default();
 
-        for (raw_state_transition, state_transition) in valid_state_transitions {
-            tracing::trace!(?state_transition, "Processing state transition");
+        for decoded_state_transition in state_transition_container.into_iter() {
```

Comment:
There is no point to refactor here away from the destructure. If you want to do that you need to do it in it's own PR as it is refactoring. We used to do valid state transitions first, we can't at this point just refactor to change the logic without an explanation as to why this change is good.

Reply:
Explained it on the call. Agreed that we need additional comments in code and PR description